feature/google/src/main/java/com/revenuecat/purchases/google/BillingWrapper.kt

```diff
@@ -797,25 +772,83 @@ class BillingWrapper(
     }
 
     private fun buildPurchaseParams(
-        storeProduct: GoogleStoreProduct,
-        purchaseOption: GooglePurchaseOption,
+        storeProduct: StoreProduct,
+        purchaseOption: PurchaseOption?,
         replaceSkuInfo: ReplaceSkuInfo?,
-        appUserID: String,
-        callback: ()
-    ): BillingFlowParams {
+        appUserID: String
+    ): Result<BillingFlowParams, PurchasesError> {
+        val googleProduct = storeProduct.googleProduct
+
+        if (googleProduct == null) {
+            val error = PurchasesError(
+                PurchasesErrorCode.UnknownError,
+                PurchaseStrings.INVALID_STORE_PRODUCT_TYPE.format(
+                    "Play",
+                    "GoogleStoreProduct"
+                )
+            )
+            errorLog(error)
+            return Result.Error(error)
+        }
+
+        return if (googleProduct.type == ProductType.SUBS) {
+            buildSubscriptionPurchaseParams(googleProduct, purchaseOption, replaceSkuInfo, appUserID)
+        } else {
+            buildOneTimePurchaseParams(googleProduct, appUserID)
+        }
+    }
+
+    private fun buildOneTimePurchaseParams(
+        googleProduct: GoogleStoreProduct,
+        appUserID: String
+    ): Result<BillingFlowParams, PurchasesError> {
         val productDetailsParamsList = BillingFlowParams.ProductDetailsParams.newBuilder().apply {
-            setOfferToken(purchaseOption.token)
-            setProductDetails(storeProduct.productDetails)
+            setProductDetails(googleProduct.productDetails)
```

Comment:
For OTP, we don't have to set the `token`. I didn't find much documentation about this from Google, so we should test this flow thoroughly once we have the sample app running.